### PR TITLE
Map Last Name to Custom Field

### PR DESCRIPTION
**Pull request by n7studios**
## Summary

Adds an option to store a MemberMouse Member's last name in a Kit Custom Field for #5 

Screenshot 2024-11-26 at 15 35 36

Other data can be supported (billing address, shipping address, phone number) in future PR's, if required - similar to how custom field data is mapped in WooCommerce.

## Testing

- `BundleCustomFieldsCest`: Test that the last name is stored in the subscriber's custom field data in Kit when assigning a bundle to a member.
- `MemberCustomFieldsCest`: Test that the last name is stored in the subscriber's custom field data in Kit when assigning a membership level to a member.
- `ProductCustomFieldsCest`: Test that the last name is stored in the subscriber's custom field data in Kit when assigning a product to a member.
- `testMemberNameAndEmailUpdatedWhenChangedInMemberMouse`: Updated to include testing custom field data is stored in Kit against the subscriber when updating a member in MemberMouse
- `testSaveCustomFieldsAssignment`: Test that the settings UI saves the custom field section when changed

## Checklist

* [x] I have [written a test](https://github.com/ConvertKit/convertkit-wordpress/blob/main/TESTING.md#writing-an-acceptance-test) and included it in this PR
* [x] I have [run all tests](https://github.com/ConvertKit/convertkit-wordpress/blob/main/TESTING.md#run-tests) and they pass
* [x] The code passes when [running the PHP CodeSniffer](https://github.com/ConvertKit/convertkit-wordpress/blob/main/TESTING.md#run-php-codesniffer)
* [x] Code meets [WordPress Coding Standards](https://github.com/ConvertKit/convertkit-wordpress/blob/main/DEVELOPMENT.md#coding-standards) for PHP, HTML, CSS and JS
* [x] [Security and Sanitization](https://github.com/ConvertKit/convertkit-wordpress/blob/main/DEVELOPMENT.md#security-and-sanitization) requirements have been followed
* [x] I have assigned a reviewer or two to review this PR (if you're not sure who to assign, we can do this step for you)

### CODE REVIEW

File: includes/class-convertkit-mm-actions.php [hunk 1]
```diff
@@ -78,17 +78,18 @@ public function add_member( $member_data ) {
 		}
 
 		// Fetch data from member array.
-		$user_email = $member_data['email'];
-		$first_name = rawurlencode( $member_data['first_name'] );
```
Comment (n7studios) [inline includes/class-convertkit-mm-actions.php:82]: We don't need to `rawurlencode` names, as the API class will handle encoding spaces.

File: tests/acceptance/bundle/BundleCustomFieldsCest.php [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+		// Check that the subscriber has been assigned to the tag.
+		$I->apiCheckSubscriberHasTag($I, $subscriber['id'], $_ENV['CONVERTKIT_API_TAG_ID']);
+
+		// Check that the subscriber has the custom field data.
+		$I->apiCustomFieldDataIsValid(
+			$I,
+			$subscriber,
+			[
+				'last_name' => 'Last',
```
Comment (noelherrick) [inline tests/acceptance/bundle/BundleCustomFieldsCest.php:72]: Where is this value coming from?

**Review by noelherrick — APPROVED** [review]
LGTM - this was a 1k line PR but it contained changes from two other PRs that I recognized. Would you be able to streamline your PRs in the future? I would appreciate it 😄

File: tests/acceptance/bundle/BundleCustomFieldsCest.php (hunk 2 above)
Comment (n7studios) [reply tests/acceptance/bundle/BundleCustomFieldsCest.php:72]: It's set when creating a new member in MemberMouse:
https://github.com/Kit/convertkit-membermouse/blob/last-name-custom-field/tests/_support/Helper/Acceptance/MemberMouse.php#L114

**Pull request merged**